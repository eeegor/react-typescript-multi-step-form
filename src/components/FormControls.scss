.form-controls {
	position: relative;
	display: flex;
	padding: 3rem 0;

	svg {
		display: flex;
		margin-left: -0.5rem;
	}
}
