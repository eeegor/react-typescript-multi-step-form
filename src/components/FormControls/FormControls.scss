.form-controls {
	position: relative;
	display: flex;
	padding: 1rem 0;

	@media (min-width: 767px) {
		padding: 3rem 0;
	}

	.icon-back {
		display: flex;
		margin-left: -0.5rem;
	}
}
