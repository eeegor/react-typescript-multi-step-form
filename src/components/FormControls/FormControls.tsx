/**
 * Component FormControls
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { IconBack, FormProgress } from '..';
import './FormControls.scss';

export interface Props {
	className?: string;
	maxSteps: number;
	currentStep: number;
	onBack: (event: React.MouseEvent<HTMLElement>) => void;
}

export class FormControls extends React.Component<Props> {
	render(): JSX.Element {
		const { className, onBack, maxSteps, currentStep } = this.props;

		return (
			<div className={classnames('form-controls', className)}>
				<div className="form-controls__back" onClick={event => onBack && onBack(event)}>
					<IconBack id="icon-back" />
				</div>
				<FormProgress maxSteps={maxSteps} currentStep={currentStep} />
			</div>
		);
	}
}
