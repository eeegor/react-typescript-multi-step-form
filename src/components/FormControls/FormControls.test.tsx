import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { FormControls } from './FormControls';

describe('FormControls', () => {
	it('renders markup correctly', () => {
		const callback = jest.fn();
		const wrapper = <FormControls maxSteps={3} currentStep={1} onBack={callback} />;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('click event works', () => {
		const callback = jest.fn();
		const wrapper = mount(<FormControls maxSteps={3} currentStep={2} onBack={callback} />);
		wrapper.find('.form-controls__back').simulate('click');
		expect(callback).toHaveBeenCalledTimes(1);
	});
});
