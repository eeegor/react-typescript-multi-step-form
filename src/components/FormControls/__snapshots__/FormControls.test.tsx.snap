// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControls renders markup correctly 1`] = `
<div
  className="form-controls"
>
  <div
    className="form-controls__back"
    onClick={[Function]}
  >
    <svg
      className="icon-back"
      height="36"
      id="icon-back"
      version="1.1"
      viewBox="0 0 512 512"
      width="36"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        fill="#eee"
        points="352,128.4 319.7,96 160,256 160,256 160,256 319.7,416 352,383.6 224.7,256 "
      />
    </svg>
  </div>
  <div
    className="form-progress"
  >
    <div
      className="form-progress__indicator-bar"
      style={
        Object {
          "width": "0%",
        }
      }
    />
    <div
      className="form-progress__background-bar"
    />
  </div>
</div>
`;
